app/cdap/components/WranglerV2/SearchField/index.tsx

```diff
@@ -0,0 +1,155 @@
+  link: string;
+  usage: string;
+}
+
+export interface ISearchFieldProps {
+  onPostfixIconClick: () => void;
+  inputProps: InputProps;
+  directiveSuggestion: IDirectiveSuggestionProps;
+}
+
+export const InputWrapper = styled(Input)`
+  &&& {
+    width: 100%;
+    margin: 0px 5px;
+    outline: 0;
+    background: transparent;
+    color: ${grey[50]};
+    & .MuiInputBase-input {
+      padding: 0px;
+      color: ${grey[50]};
+      font-size: 14px;
+    }
+  }
+`;
+
+const PrefixIconWrapper = styled(Box)`
+  color: ${lightGreen[400]};
+`;
+
+const PostfixIconWrapper = styled(CloseOutlinedIcon)`
+  cursor: pointer;
+`;
+
+const DirectiveWrapper = styled(Box)`
+  padding: 0px 15px;
+`;
+
+const DividerLine = styled(Divider)`
+  &.MuiDivider-root {
+    background-color: ${grey[300]};
+    opacity: 0.2;
+    margin-bottom: 10px;
+  }
+`;
+
+const InfoLink = styled(Link)`
+  &&& {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    color: ${blue.A100};
+    font-size: 14px;
+    text-decoration: none;
+    margin-left: 35px;
+    width: 217px;
+    cursor: pointer;
+  }
+`;
+
+const StyledInfoOutlinedIcon = styled(InfoOutlinedIcon)`
+  &&& {
+    color: ${blue.A100};
+    width: 20px;
+    height: 20px;
+  }
+`;
+
+const DirectivesUsageText = styled(Typography)`
+  &&& {
+    font-style: normal;
+    font-weight: 400;
+    font-size: 14px;
+    letter-spacing: 0.15;
+    color: ${grey[50]};
+    margin-bottom: 15px;
+    display: flex;
+    align-items: center;
+    width: 100%;
+  }
+`;
+
+const DirectiveFieldWrapper = styled(Box)`
+  &&& {
+    background: ${grey[700]};
+    padding: 20px;
+    color: ${grey[50]};
+    min-height: 82px;
+  }
+`;
+
+const SearchFieldWrapper = styled(Box)`
+  &&& {
+    display: flex;
+    font-size: 14px;
+    margin-bottom: 10px;
+  }
+`;
+
+const PREFIX = 'features.WranglerNewUI.DirectivePanel';
+
+export default function SearchField({
+  onPostfixIconClick,
+  inputProps,
+  directiveSuggestion,
+}: ISearchFieldProps) {
+  return (
+    <DirectiveFieldWrapper>
+      {directiveSuggestion?.usage && (
+        <DirectiveWrapper data-testid="directive-usage-text-wrapper">
+          <DirectivesUsageText variant="body1" data-testid="directive-usage-text">
+            {`${T.translate(`${PREFIX}.usage`)} : `}
+            {directiveSuggestion.usage}
+            {directiveSuggestion.link && (
+              <InfoLink data-testid="info-link" href={directiveSuggestion.link} target="_blank">
+                <StyledInfoOutlinedIcon data-testid="info-icon" />
+                {T.translate(`${PREFIX}.moreInfoOnDirective`)}
+              </InfoLink>
+            )}
+          </DirectivesUsageText>
+          <DividerLine />
+        </DirectiveWrapper>
+      )}
+      <SearchFieldWrapper>
+        <PrefixIconWrapper>$</PrefixIconWrapper>
+        <InputWrapper {...inputProps} disableUnderline={true} />
+        <PostfixIconWrapper onClick={onPostfixIconClick} />
```

Comment:
we should not add onClick or other events on icons, divs, etc. wrap the icon inside MUI IconButton and give onClick on that button
plz update this across the application